typescript/sdk/src/token/types.ts

```diff
@@ -166,3 +222,109 @@ function isCollateralRebasePairedCorrectly(
+  ccipIsmMap: CCIPContractExistsMap;
+} {
+  const ccipHookMap: CCIPContractExistsMap = {};
+  const ccipIsmMap: CCIPContractExistsMap = {};
+
+  Object.entries(warpRouteDeployConfig).forEach(([chainName, config]) => {
+    extractCCIPHookMap(chainName, config.hook, ccipHookMap);
+    extractCCIPIsmMap(chainName, config.interchainSecurityModule, ccipIsmMap);
+  });
+
+  return { ccipHookMap, ccipIsmMap };
+}
+
+function extractCCIPHookMap(
+  currentChain: ChainName,
+  hookConfig: HookConfig | undefined,
+  existsCCIPHookMap: CCIPContractExistsMap,
+) {
+  if (!hookConfig || typeof hookConfig === 'string') {
+    return;
+  }
+
+  switch (hookConfig.type) {
+    case HookType.AGGREGATION:
+      hookConfig.hooks.forEach((hook) =>
+        extractCCIPHookMap(currentChain, hook, existsCCIPHookMap),
+      );
+      break;
+    case HookType.ARB_L2_TO_L1:
+      extractCCIPHookMap(currentChain, hookConfig.childHook, existsCCIPHookMap);
+      break;
+    case HookType.CCIP:
+      if (!existsCCIPHookMap[currentChain]) {
+        existsCCIPHookMap[currentChain] = new Set();
+      }
+      existsCCIPHookMap[currentChain].add(hookConfig.destinationChain);
+      break;
+    case HookType.FALLBACK_ROUTING:
+    case HookType.ROUTING:
+      Object.entries(hookConfig.domains).forEach(([_, hook]) => {
+        extractCCIPHookMap(currentChain, hook, existsCCIPHookMap);
+      });
+      break;
+    case HookType.INTERCHAIN_GAS_PAYMASTER:
+    case HookType.MERKLE_TREE:
+    case HookType.OP_STACK:
+    case HookType.PAUSABLE:
+    case HookType.PROTOCOL_FEE:
+      break;
+  }
+}
+
+function extractCCIPIsmMap(
+  currentChain: ChainName,
+  ismConfig: IsmConfig | undefined,
+  existsCCIPIsmMap: CCIPContractExistsMap,
+) {
+  if (!ismConfig || typeof ismConfig === 'string') {
+    return;
+  }
+
+  switch (ismConfig.type) {
+    case IsmType.AGGREGATION:
+    case IsmType.STORAGE_AGGREGATION:
+      ismConfig.modules.forEach((hook) =>
+        extractCCIPIsmMap(currentChain, hook, existsCCIPIsmMap),
+      );
+      break;
+    case IsmType.CCIP:
+      if (!existsCCIPIsmMap[ismConfig.originChain]) {
+        existsCCIPIsmMap[ismConfig.originChain] = new Set();
+      }
+      existsCCIPIsmMap[ismConfig.originChain].add(currentChain);
+      break;
+    case IsmType.FALLBACK_ROUTING:
+    case IsmType.ROUTING:
+      Object.entries(ismConfig.domains).forEach(([_, hook]) => {
+        extractCCIPIsmMap(currentChain, hook, existsCCIPIsmMap);
+      });
+      break;
+    case IsmType.ARB_L2_TO_L1:
+    case IsmType.ICA_ROUTING:
+    case IsmType.MERKLE_ROOT_MULTISIG:
+    case IsmType.MESSAGE_ID_MULTISIG:
+    case IsmType.PAUSABLE:
+    case IsmType.OP_STACK:
+    case IsmType.STORAGE_MERKLE_ROOT_MULTISIG:
+    case IsmType.STORAGE_MESSAGE_ID_MULTISIG:
+    case IsmType.TEST_ISM:
+    case IsmType.TRUSTED_RELAYER:
+    case IsmType.WEIGHTED_MERKLE_ROOT_MULTISIG:
+    case IsmType.WEIGHTED_MESSAGE_ID_MULTISIG:
+      break;
+  }
+}
```

Comment:
I believe this could be replaced with `deepFind` and not account for every enum variant